src/vs/gitpod/browser/workbench/workbench.html

```diff
@@ -3,6 +3,74 @@
 <html>
 	<head>
 		<script>
+			async function gitpodMetricsAddCounter(metricsName, labels, value) {
+				function getMetricsUrl() {
+					const baseWorkspaceIDRegex = "(([a-f][0-9a-f]{7}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})|([0-9a-z]{2,16}-[0-9a-z]{2,16}-[0-9a-z]{8,11}))";
+					// this pattern matches URL prefixes of workspaces
+					const workspaceUrlPrefixRegex = RegExp(`^([0-9]{4,6}-)?${baseWorkspaceIDRegex}\\.`);
+					let url = new URL(window.location.href)
+					url.search = "";
+					url.hash = "";
+					url.pathname = "";
+					if (url.host.match(workspaceUrlPrefixRegex)) {
+						url.host = url.host.split(".").splice(2).join(".")
+					} else {
+						return
+					}
+					const hostSegments = url.host.split(".");
+					if (hostSegments[0] !== "ide") {
+						url.host = "ide." + url.host
+					}
+					url.pathname = "/metrics-api";
+					return url.toString();
+				}
+				try {
+					const MetricsUrl = getMetricsUrl()
+					if (!MetricsUrl) {
+						return false
+					}
+					const url = `${MetricsUrl}/metrics/counter/add/${metricsName}`
```

Comment:
suggestion(nit): don't capitalize the variable name [[1](https://stackoverflow.com/a/1564489/10199319)]
```suggestion
					const metricsUrl = getMetricsUrl()
					if (!metricsUrl) {
						return false
					}
					const url = `${metricsUrl}/metrics/counter/add/${metricsName}`
```